src/plugins/git/mergeRepository.js

```diff
@@ -0,0 +1,18 @@
+// @flow
+
+import type {Repository} from "./types";
+
+export function mergeRepository(
+  repositories: $ReadOnlyArray<Repository>
+): Repository {
+  const newRepository = {commits: {}, trees: {}};
+  for (const {trees, commits} of repositories) {
+    for (const treeHash in trees) {
```

Comment:
Looks like `for`-`in` is safe when the object in question is just a struct; if you have inherited/non-enumerable properties, it’s less so. This is not technically guaranteed for us because the `commits` and `trees` are not exact. But it should be safe with our existing code thus far. Up to you whether to convert to `Object.keys`.

Reply:
Done.